torch/_dynamo/optimizations/analysis.py

```diff
@@ -126,9 +126,22 @@ def has_mutation(gm, example_inputs, inputs_only=False):
     # like "leaf variable that requires grad is inplace modified"
     example_inputs = clone_inputs(example_inputs)
     if fake_tensors_available and config.fake_tensor_propagation:
-        with FakeTensorMode() as fake_mode:
+
+        def _wrap_to_fake_tensor(t, *, f_mode):
+            if type(t) in (torch.Tensor, torch.nn.Parameter):
+                static_shapes_ = config.dynamic_shapes is False
+                return fake_mode.from_tensor(t, static_shapes=config.dynamic_shapes is not False)
+            else:
+                return t
+
+        # Our analysis pass should use dynamic shape tensor inputs
+        # when dynamic shapes are enabled.
+        # We don't actually care about the guards that are created
+        # on those shapes though, so just create a fresh ShapeEnv here.
+        from torch.fx.experimental.symbolic_shapes import ShapeEnv
```

Comment:
We should really land the input mutation AOTAutograd PR and kill this analysis pass.

The problem here was that dynamo is adding nodes like `floordiv = getitem_4.__floordiv__(7)`. That's fine when `getitem_4` is a `SymFloat`, but plain old python `float` doesn't have a `__floordiv__` method.

I guess an alternative would be to get dynamo to bake in functions instead of methods, and overload all of the methods to do the right thing for ints/floats. But it felt like the more correct thing was that if we originally traced a program with symbolic shapes, our later analysis passes should use symbolic shapes too.

Reply:
Did you have a chance to read https://docs.google.com/document/d/19UoIh_SVrMy_b2Sx5ZaeOJttm6P0Qmyss2rdBuyfoic/edit ? What did you think?

Reply:
Started reading, still working through it! 😛